test/test_nn.py

```diff
@@ -68,7 +68,45 @@
 
 DOUBLE_TENSORTYPES = [torch.double]
 
-
```

Comment:
Should add a comment explaining what these test cases are for, and what their format is.